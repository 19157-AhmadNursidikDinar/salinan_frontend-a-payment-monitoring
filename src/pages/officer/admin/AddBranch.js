import React from "react";
//@material-ui/core
import {
    Container,
    TextField,
    Button,
    Grid,
    Typography,
    Paper,
} from "@material-ui/core";
import { makeStyles } from '@material-ui/core/styles';
//@material-ui/icons
import SaveRoundedIcon from "@material-ui/icons/SaveRounded";
import ArrowBackIosRounded from "@material-ui/icons/ArrowBackIosRounded";
//links
import { Link } from "react-router-dom";
import ContentContainer from "../../../components/ContentContainer";
//Styling Page
const useTheStyle = makeStyles((theme) => ({
PaperSize: {
        padding: 40,
    },
    BtnSave: {
        backgroundColor: "#1890FF",
        '&:hover': {
            backgroundColor: "# 2979ff",
        },
        marginTop: 30,
        float: 'right',
    },
    BtnBack: {
        backgroundColor: "#1890FF",
        '&:hover': {
            backgroundColor: "# 2979ff",
        },
        marginTop: 30,
        float: 'left',
    },
}));

function FormAddBranch() {
    const classes = useTheStyle();

    //page add branch
    return(
        <Paper>
            <Container>
                <Grid container spacing={3}>
                    <Grid item xs={12}>
                        <TextField 
                            id="BranchName"
                            name="BranchName"
                            label="Nama Kantor Cabang"
                            variant="outlined"
                            fullWidth />
                    </Grid>
                </Grid>
            </Container>
            <Grid container spacing={3}>
                <Grid item xs={12}>
                    <Link to="/branch-office-list">
                        <Button 
                            variant="contained"
                            color="primary"
                            className={classes.BtnBack}
                            startIcon={<ArrowBackIosRounded />}
                            >  
                            Kembali
                        </Button>
                    </Link>
                    <Link to="/branch-office-list">
                        <Button 
                            variant="contained"
                            color="primary"
                            className={classes.BtnSave}
                            SaveRounded={<SaveRoundedIcon />}
                            >
                            Simpan
                        </Button>
                    </Link>
                </Grid>
            </Grid>
        </Paper>
    );
}

export default function AddBranch() {
    return (
        <ContentContainer role="admin" selectedMenu="Daftar Kantor Cabang">
            <div
                style={{
                    display: "flex",
                    justifyContent: "center",
                    width: "100%",
                    paddingBottom: "2em",
                }}
            >
                <Typography variant="h4">Tambah Kantor Cabang</Typography>
            </div>
            <FormAddBranch />
        </ContentContainer>
    );
}
