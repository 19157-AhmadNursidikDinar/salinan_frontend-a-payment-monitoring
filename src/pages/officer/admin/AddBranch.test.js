import React from "react";
import { shallow } from "enzyme";
import AddBranch from "./AddBranch";
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";

describe("Add Branch", () => {
    const wrapper = shallow(<AddBranch />);
    it("should render Add Branch", () =>{
        expect(wrapper).toMatchSnapshot();
    });
    it("should render TextField", () =>{
        expect(wrapper.find(TextField).length).toBe(1);
    });
})

describe("Test Button Add", () => {
    it("Test click event", () => {
        const mockCallBack = jest.fn();
        const button = shallow((<Button onClick={mockCallBack}>Success!</Button>));
        button.find("button").simulate("click");
        expect(mockCallBack.mock.calls.length).toEqual(1);
    })
})