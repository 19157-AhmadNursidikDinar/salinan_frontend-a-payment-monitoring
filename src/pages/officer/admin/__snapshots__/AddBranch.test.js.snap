// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Branch should render Add Branch 1`] = `ShallowWrapper {}`;
